web/src/SidebarResources.test.tsx

```diff
@@ -132,8 +135,108 @@ describe("SidebarResources", () => {
     expectIncr(1, "ui.web.pin", { pinCount: "1", action: "unpin" })
 
-    expect(localStorage.getItem(makeKey("test", "pinned-resources"))).toEqual(
-      JSON.stringify(["vigoda"])
-    )
+    expect(pinnedItemsAccessor.get()).toEqual(["vigoda"])
   })
+
+  const loadCases: [string, SidebarOptions, string[]][] = [
+    [
+      "showResources",
+      { showResources: false, showTests: true, alertsOnTop: true },
+      ["a", "b"],
+    ],
+    [
+      "showTests",
+      { showResources: true, showTests: false, alertsOnTop: true },
+      ["vigoda"],
+    ],
+    [
+      "alertsOnTop",
+      { showResources: true, showTests: true, alertsOnTop: false },
+      ["vigoda", "a", "b"],
+    ],
+  ]
+  test.each(loadCases)(
+    "loads %p from localStorage",
+    (name, options, expectedItems) => {
+      sidebarOptionsAccessor.set(options)
+
+      const items = [
+        oneResource(),
+        oneResourceTestWithName("a"),
+        oneResourceTestWithName("b"),
+      ].map((res) => new SidebarItem(res))
+
+      const root = mount(
+        <MemoryRouter>
+          <tiltfileKeyContext.Provider value="test">
+            <SidebarResources
+              items={items}
+              selected={""}
+              resourceView={ResourceView.OverviewDetail}
+              pathBuilder={pathBuilder}
+            />
+          </tiltfileKeyContext.Provider>
+        </MemoryRouter>
+      )
+
+      assertSidebarItemsAndOptions(
+        root,
+        expectedItems,
+        options.showResources,
+        options.showTests,
+        options.alertsOnTop
+      )
+    }
+  )
+
+  const saveCases: [string, SidebarOptions][] = [
+    [
+      "showResources",
+      { showResources: false, showTests: true, alertsOnTop: true },
+    ],
+    ["showTests", { showResources: true, showTests: false, alertsOnTop: true }],
+    [
+      "alertsOnTop",
+      { showResources: true, showTests: true, alertsOnTop: false },
+    ],
+  ]
+  test.each(saveCases)(
+    "saves option %s to localStorage",
+    (name, expectedOptions) => {
+      const items = [
+        oneResource(),
+        oneResourceTestWithName("a"),
+        oneResourceTestWithName("b"),
+      ].map((res) => new SidebarItem(res))
+
+      const root = mount(
+        <MemoryRouter>
+          <tiltfileKeyContext.Provider value="test">
+            <SidebarResources
+              items={items}
+              selected={""}
+              resourceView={ResourceView.OverviewDetail}
+              pathBuilder={pathBuilder}
+            />
+          </tiltfileKeyContext.Provider>
+        </MemoryRouter>
+      )
+
+      root.find("input#resources").simulate("change", {
+        target: { checked: expectedOptions.showResources },
+      })
+      root
+        .find("input#tests")
+        .simulate("change", { target: { checked: expectedOptions.showTests } })
+      if (
+        root.find(AlertsOnTopToggle).hasClass("is-enabled") !=
```

Comment:
this should probably be `simulate("change"...)` to match the others? (maybe `simulate("click")` is sufficient for saving things to storage but not for triggering the onChange handler? eh I don't know. You can also leave this as-is and I can fix it if need be when I go back and write AlertsOnTop tests.)

Reply:
The others are `CheckBox`es and define `onChange` handlers, while `AlertsOnTopToggle` is a `Button` and defines an `onClick` handler. It would be nice if they were more consistent, but I think that's a code change rather than a test change, and more than I want to take on in this PR.